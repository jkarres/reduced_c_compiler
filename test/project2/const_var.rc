// testing using const vars in exprs

const int gci = 23 + 1;

const float gcf = 9.22 + 1;

const bool gcb = !true;

function : void main() {
  const int lci = 24;
  const float lcf = 10.33;
  const bool lcb = !false;

  cout << 1 << " " << 1 + gci << endl;
  cout << 2 << " " << 2.2 + gcf << endl;
  cout << 3 << " " << false || gcb << endl;

  cout << 4 << " " << lci - 1 << endl;
  cout << 5 << " " << lcf - 3.4 << endl;
  cout << 6 << " " << true && lcb << endl;
}
  