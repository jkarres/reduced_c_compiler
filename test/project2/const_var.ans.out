1 25
2 12.42
3 false
4 23
5 6.93
6 true
